dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -120,7 +137,24 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     {
         if (IsPackagedAppScenario())
         {
-            m_channel.PushNotificationReceived(token);
+            if (m_channel)
+            {
+                m_channel.PushNotificationReceived(token);
+            }
+            else
+            {
+                auto lock = m_lock.lock_exclusive();
+
+                if (!--m_foregroundHandlerCount)
```

Comment:
Looks like the counter here will have the similar problem as Jon mentioned in the counter adding case. can we move the count operation after the operations?